### PR TITLE
feat: use a new official CH driver: clickhouse-connect

### PR DESCRIPTION
**Pull request by EugeneTorap**
Use official `clickhouse-connect` driver instead of `clickhouse-sqlalchemy`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — COMMENTED** [review]
Excited about this development, thanks for the PR 👍  One question about upgrading to 0.4.0 (it worked well locally for me). Also, could we migrate the legacy tests from `tests/integration_tests/db_engine_specs/clickhouse_tests.py` to `tests/unit_tests/db_engine_specs/test_clickhouse.py` and make sure all relevant code paths are covered? If you need help please ping me on Slack.

File: docs/docs/databases/clickhouse.mdx [hunk 1]
```diff
@@ -10,33 +10,33 @@ version: 1
...
+clickhousedb+connect://demo:demo@github.demo.trial.altinity.cloud/default?secure=true
 ```
 
-If you're using Clickhouse locally on your computer, you can get away with using a native protocol URL that
+If you're using Clickhouse locally on your computer, you can get away with using a http protocol URL that
 uses the default user without a password (and doesn't encrypt the connection):
 
 ```
-clickhouse+native://localhost/default
+clickhousedb+connect://localhost/default
```
Comment (genzgd) [inline docs/docs/databases/clickhouse.mdx:41]: `+connect` is not required

**Comment by EugeneTorap** [comment]
Waiting for the 0.4.1 version of `clickhouse-connect`

**Comment by EugeneTorap** [comment]
@villebro @genzgd How to fix the next error?

> test_clickhouse.py:42: error: Unexpected keyword argument "message" for "NewConnectionError"

**Comment by genzgd** [comment]
I think tests are little tricky because `clickhouse-connect` provides its own EngineSpec class that is not part of the superset package or code base.  As far as I know this is the only EngineSpec that works this way, and it uses the Python "entry point" mechanism to do that.

The result is the class to test is `clickhouse_connect.cc_superset.engine.ClickHouseEngineSpec` instead of `superset.db_engine_specs.clickhouse` -- `superset.db_engine_specs.clickhouse` is coupled with the clickhouse-sqlachemy library.

**Comment by genzgd** [comment]
ClickHouse Connect 0.4.1 has been released on PyPI.

**Comment by EugeneTorap** [comment]
> ClickHouse Connect 0.4.1 has been released on PyPI.  

Thank you so much!

**Review by villebro — APPROVED** [review]
LGTM and tested to work nicely. Looking forward to using this one in the future! 👍

**Pull request merged**